File: vlantana-wms-backend/WMS.Web/Models/Business/Product.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace vlantana_wms_backend.Models.Business
{
    public record class Product
    {
        [Key]
        public int ProductId { get; set; }

        [Required]
        [StringLength(50)]
        public string? Name { get; set; }

        [Required]
        [StringLength(50)]
        public string? Description { get; set; }

        [Required]
        public string? Status { get; set; }

        [Required]
        public DateTime ProductionDate { get; set; }

        [Required]
        public DateTime ExpiryDate { get; set; }

        // Foreign key
        public int? PalleteId { get; set; }
        public int? OrderId { get; set; }
        public int? CompanyId {  get; set; }

        // Belongs to
        public virtual Pallete? Pallete { get; set; }
        public virtual Order? Order { get; set; }
        public virtual Company? Company { get; set; }

    }
}

File: vlantana-wms-backend-laravel/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Product extends Model
{
    use HasFactory;

    protected $fillable = [
        'pallet_id',
        'order_id',
        'company_id',
        'name',
        'description',
        'status',
        'production_date',
        'expiry_date',
    ];

    public function pallet() : BelongsTo {
        return $this->belongsTo(Pallet::class);
    }

    public function order() : BelongsTo {
        return $this->belongsTo(Order::class);
    }

    public function company() : BelongsTo {
        return $this->belongsTo(Company::class);
    }
}
